sda/cmd/s3inbox/proxy.go

```diff
@@ -463,32 +457,20 @@ func (p *Proxy) CreateMessageFromRequest(r *http.Request, claims jwt.Token) (Eve
 // the etag and size information for the uploaded document
 func (p *Proxy) requestInfo(fullPath string) (string, int64, error) {
 	filePath := strings.Replace(fullPath, "/"+p.s3.Bucket+"/", "", 1)
-	s, err := p.newSession()
+	client, err := storage.NewS3Client(p.s3)
 	if err != nil {
 		return "", 0, err
 	}
-	svc := s3.New(s)
+
 	input := &s3.ListObjectsV2Input{
-		Bucket:  aws.String(p.s3.Bucket),
-		MaxKeys: aws.Int64(1),
-		Prefix:  aws.String(filePath),
+		Bucket:  &p.s3.Bucket,
+		MaxKeys: aws.Int32(1),
+		Prefix:  &filePath,
 	}
 
-	result, err := svc.ListObjectsV2(input)
+	result, err := client.ListObjectsV2(context.TODO(), input)
```

Comment:
why not `context.Background` https://pkg.go.dev/context#Background ?

Reply:
According to the doc, when you need a context but you don't have one (yet) and don't know what to use, use context.TODO().

Reply:
ok, i understood those are similar, just some semantic differences. I am fine with other

```
Background()` function provides a root context for use when no other context is available. The `context. TODO()` function serves as a placeholder for when it is unclear which context to use
```
https://reintech.io/term/context-background-and-context-todo-in-go